### ISSUE TITLE
Input Gestures Dialog: Don't allow Duplicate Gestures to be Defined

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10872]
### Steps to reproduce:

For this example, we'll use the "Read Current Status Bar" action

1. Open the Input Gestures dialog, and find the "Read Current Status Bar" action

2. Hit the "Add" button, and when prompted to specify the gesture to use, hit NVDA + S; specify that it should be active for both Desktop and Laptop keyboard layouts

3. Once again, hit the "Add" button, and when prompted to perform the gesture, specify the exact same gesture as before (NVDA + S) and choose to bind the gesture to both the Desktop and Laptop keyboard layouts - when done, you'll notice two duplicate gestures in the list for the action

4. Hit the "OK" button to save the changes - when now looking at gestures.ini, you'll notice that the new gesture has been added twice

5. Re-open the Input Gestures dialog, and find the "Read Current Status Bar" action - you'll notice that three gestures are defined: the two default gestures that ship with NVDA, and the custom gesture that we defined previously

6. Highlight our custom gesture, and hit the "Remove" button to remove it from the list - next, hit the "OK" button

7. Go back into the Input Gestures dialog, and once again find "Read Current Status Bar" - despite just removing our custom gesture, you'll notice that it is still listed. Hit "Remove" once again, and then finally hit "OK" to remove it once and for all 

The above behaviour can be observed because it appears that NVDA doesn't check to see if the gesture already exists when a gesture is added; as a result, when we chose to remove our custom gesture the first time, we only removed a single instance of it from gestures.ini; when opening the Input Gestures dialog again the gesture was still listed due to the final instance still being present within the gestures.ini file.

### Expected behavior:

When adding a gesture, NVDA should first check to ensure that the same gesture hasn't already been bound to the exact same action as the one that the user is attempting to define

#### NVDA version:

2019.3.1

**Comment by DrSooom** [comment]
Please restart NVDA after step 4. I guess an error message still prompts on NVDA startup. Then open "gestures.ini" with Notepad++. This file should now contain two identical entries. I played with this – I guess – in 2018. So I'm unsure about the current situation regarding duplicate allocations.

**Comment by ehollig** [comment]
This is related to #7074

**Comment by Adriani90** [comment]
Closing as duplicate of #3848.

**Comment by XLTechie** [comment]
@Adriani90 I object to this being closed.
It may be better listed as a duplicate of #3605, but even so, it highlights an 
even stranger issue than either of those. If you look at the original STR, it 
demonstrates assigning one gesture twice to the same action. That is not covered 
by either of the other issues, and should not be permitted at all.

It also requires a slightly different fix than either #3605 or #3848.

**Comment by CyrilleB79** [comment]
Reopening as per https://github.com/nvaccess/nvda/issues/10872#issuecomment-2024175437.

**Comment by theshippeys** [comment]
I took a look at this as a first step to contributing to NVDA, since it was tagged `good first issue`. I checked if the bug could be recreated at SHA `f56b2e0d7`, since the original ticket is over 6 years old.

There seem to be 2 slightly different behaviours, depending on whether the duplicate is added in the same `InputGesturesDialog` invocation as the original
1. New dialog instance - as described in the ticket. The duplicate is not spotted, but duplicate causes odd behaviour 
2. Same dialog instance - the Input Gestures Dialog can't be opened again until NVDA is restarted. The [logging](https://github.com/user-attachments/files/32605578/log.txt) shows the unhandled `ValueError` exception leaves a stale `InputGesturesDialog` instance which stops any new instance being created. This seems the less likely case, but I think the behaviour would confuse a user.

I have a potential fix which handles the exception by displaying a Message Box, and putting the dialog back in a state to accept a new gesture. It also broadens the check for a duplicate so case 1 is handled in the same way as case 2. Can I create a PR for this? 

I see from https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/contributing.md that for non minor bug fix issues, they should be labelled as `triaged` prior to being worked. I guess this is classed a minor bug fix?

**Comment by CyrilleB79** [comment]
@seanbudd yould you mind triage this one?

@theshippeys if you already have a branch, you can submit it as a PR, with the risk that NV Access does not accept your fix design. If you haven't implemented anything yet, you'd better wait for @seanbudd's triage and potential comments before proceeding.
In any case, if you are using AI, beware to respect NV Access AI policy.

**Comment by theshippeys** [comment]
@CyrilleB79 I've got the change on a branch, so I'll submit a PR, if that's OK. Obviously happy to update the solution with any and all feedback, for a first change on a new codebase! 

And if it's triaged prior to the PR being submitted, then will update the PR locally with the recommended approach.
